SCF-Session-6/polynomial/Poly.java

```diff
@@ -0,0 +1,225 @@
+     * @param value, value of variable
+     * @return the result after evaluation, throws AssertionError if value is infinity
+     */
+    public double evaluate(float value)
+    {
+    	double result = 0;
+        
+        for(int i = 0; i < polynomial.length; i++)
+        {
+            result += polynomial[i][COEFFICIENT] * Math.pow(value, polynomial[i][POWER]);
+        }
+        
+        if(result == Double.POSITIVE_INFINITY || result == Double.NEGATIVE_INFINITY)
+            throw new AssertionError("Result is too large");
+        
+        return result;
+    }
+    
+    
+    /**
+     * computes the degree of the polynomial
+     * @return degree of the polynomial
+     */
+    public int degree()
+    {
+        int degree = 0;
+        
+        for(int i = 0; i < polynomial.length; i++)
+        {
+            if(polynomial[i][POWER] > degree)
+                degree = polynomial[i][POWER];
+        }
+        
+        return degree;
+    }
+    
+    
+    /**
+     * calculates the sum of the polynomials p1 and p2
+     * @param firstPoly
+     * @param secondPoly
+     * @return the resulting polynomial after addition
+     */
+    public static Poly addPoly(Poly firstPoly, Poly secondPoly)
+    {
+        int maxDegree = (firstPoly.degree() > secondPoly.degree()) ? firstPoly.degree() : secondPoly.degree();
+        int[][] addition = new int[maxDegree + 1][COLUMN];
+        int term = 0, firstCoefficient, secondCoefficient;
+        
+        for(int i = 0; i <= maxDegree; i++)
+        {
+            firstCoefficient = firstPoly.getCoefficientOfX(i);
+            secondCoefficient = secondPoly.getCoefficientOfX(i);
+            
+            if(firstCoefficient != 0 || secondCoefficient != 0)
+            {
+                addition[term][COEFFICIENT] = firstCoefficient + secondCoefficient;
+                addition[term][POWER] = i;
+                term++;
+            }
+                
+        }
+        
+        return new Poly(addition);
+    }
+    
+    
+    /**
+     * calculates the product of polynomials p1 and p2
+     * @param firstPoly
+     * @param secondPoly
+     * @return the resulting polynomial after multiplication
+     */
+    public static Poly multiplyPoly(Poly firstPoly, Poly secondPoly)
+    {
+        int maxDegree = firstPoly.degree() + secondPoly.degree();
+        int term = 0, powerTerm, duplicateTerm;
+        int[][] multiplication = new int[maxDegree + 1][COLUMN];
+        int[][] firstPolyArray = firstPoly.getPolynomial();
+        int[][] secondPolyArray = secondPoly.getPolynomial();
+        
+        for(int i = 0; i < firstPolyArray.length; i++)
+        {
+            for(int j = 0; j < secondPolyArray.length; j++)
+            {
+                powerTerm = firstPolyArray[i][POWER] + secondPolyArray[j][POWER];
+                duplicateTerm = checkPower(multiplication, powerTerm);
+                
+                if(duplicateTerm != -1)
```

Comment:
Comment should be added if it is necessary to to use the hard coded value like `-1`.

Reply:
done